src/Microsoft.ML.LightGBM/WrappedLightGbmBooster.cs

```diff
@@ -0,0 +1,276 @@
+                        var categoricalSplit = new bool[numLeaves - 1];
+                        if (categoricalFeatureBoudaries != null)
+                        {
+                            // Add offsets to split features.
+                            for (int node = 0; node < numLeaves - 1; ++node)
+                                splitFeature[node] = categoricalFeatureBoudaries[splitFeature[node]];
+                        }
+
+                        if (numCat > 0)
+                        {
+                            var catBoundaries = Str2IntArray(kvPairs["cat_boundaries"], ' ');
+                            var catThreshold = Str2UIntArray(kvPairs["cat_threshold"], ' ');
+                            for (int node = 0; node < numLeaves - 1; ++node)
+                            {
+                                if (GetIsCategoricalSplit(decisionType[node]))
+                                {
+                                    int catIdx = (int)threshold[node];
+                                    var cats = GetCatThresholds(catThreshold, catBoundaries[catIdx], catBoundaries[catIdx + 1]);
+                                    categoricalSplitFeatures[node] = new int[cats.Length];
+                                    // Convert Cat thresholds to feature indices.
+                                    for (int j = 0; j < cats.Length; ++j)
+                                        categoricalSplitFeatures[node][j] = splitFeature[node] + cats[j] - 1;
+
+                                    splitFeature[node] = -1;
+                                    categoricalSplit[node] = true;
+                                    // Swap left and right child.
+                                    int t = leftChild[node];
+                                    leftChild[node] = rightChild[node];
+                                    rightChild[node] = t;
+                                }
+                                else
+                                {
+                                    categoricalSplit[node] = false;
+                                }
+                            }
+                        }
+                        RegressionTree tree = RegressionTree.Create(numLeaves, splitFeature, splitGain,
+                            threshold.Select(x => (float)(x)).ToArray(), defaultValue.Select(x => (float)(x)).ToArray(), leftChild, rightChild, leafOutput,
+                            categoricalSplitFeatures, categoricalSplit);
+                        res.AddTree(tree);
+                    }
+                    else
+                    {
+                        RegressionTree tree = new RegressionTree(2);
+                        var leafOutput = Str2DoubleArray(kvPairs["leaf_value"], ' ');
+                        if (leafOutput[0] != 0)
+                        {
+                            // Convert Constant tree to Two-leaf tree, avoid being filter by TLC.
```

Comment:
>TLC [](start = 93, length = 3)

another mention of this.

Reply:
There are plenty of mentions of TLC across  ML.NET codebase and we should open a separate PR and address this issue. 

---
In reply to: [197886955](https://github.com/dotnet/machinelearning/pull/392#discussion_r197886955) [](ancestors = 197886955)